src/components/select/select.scss

```diff
@@ -0,0 +1,164 @@
+@use "@/styles/colors.scss";
+@use "@/styles/variables.scss";
+@use "@/styles/font-system.scss";
+
+.select {
+    position: relative;
+    max-width : max-content;
+    min-width : 200px;
+    height: 30px;
+    // height: max-content;
+    .select--label {
+        display: flex;
+        flex-direction: column;
+        cursor: pointer;
+        position: absolute;
+        color: colors.$primary;
+        top: -14px;
+        left: 0;
+        max-width : 75%;
+        overflow:hidden;
+        display:inline-block;
+        text-overflow: ellipsis;
+        white-space: nowrap;
+
+    }
+    &:has(p) {     
+        .active--option {
+            padding: 0 8px;
+            margin: 0;
+        }
+
+       .label-inside {
+            padding: 4px 4px 0 4px;
+            background-color: colors.$surface;
+            top: -10px;
+            z-index: 1;
+        }
+    }
+    .label-inside {
+        top: calc(50% - 7px);
+        left: 10px;
+        transition: all 150ms;
+        &[data-active="true"] {
+            font-size: map-get(font-system.$font-size, small);
+            padding: 4px 4px 0 4px;
+            background-color: colors.$surface;
+            top: -10px;
+            z-index: 1;
+        }
+    }
+
+    .select-container {
+        position: relative;
+        display: flex;
+        align-items: center;
+        width: 100%;
+        height: 100%;
+        border-radius: 4px;
+        border: 1px solid colors.$secondary-900;
+        transition: all 0.15s;
+        &:focus-within {
+            border-color: colors.$primary;
+        }
+        .select--button {
+            all: unset;
+            position: absolute;
+            right: 10px;
+            top: calc(50% - 8px);
+            svg {
+                color: colors.$primary;
+                cursor: pointer;
+                transition: transform 0.15s ease-in-out;
+            }
+            .options-active {
+                transform: rotate(-180deg);
+                }
+        }
+    }
+
+    #option__container {
+        background-color: colors.$select-item-bg-light;
+        border: 1px solid colors.$select-border-color-light;
+        border-radius: 4px;
+        display: none;
+        max-height: 0;
+        width: 100% !important;
+        z-index: 1;
+        &[data-show] {
+            display: flex;
+            flex-direction: column;
+            width: fit-content;
+            min-width: min-content;
+            max-height: fit-content;
+
+        }
+        .option {
+            text-align: end;
```

Comment:
What's the reason to have it end-aligned? 🙂

Reply:
AND, Set the default one `-- Please select an option --` to center. This make it stand out a bit. And you might notice that it's not a `data` alternative per se.

Reply:
Then its 2 - 1 for left ( I'm 1 - 1 :smile:), so I'm changing to left-aligned.

I also thought about centering the default one. Tried it and it stands out and doesn't look weird.